lib/config/global.ts

```diff
@@ -40,7 +40,10 @@ export class GlobalConfig {
 
     const result = { ...config };
     for (const option of GlobalConfig.OPTIONS) {
-      GlobalConfig.config[option] = config[option];
+      GlobalConfig.config = {
+        ...GlobalConfig.config,
+        [option]: config[option],
```

Comment:
```suggestion
        option: config[option],
```

Shouldn't it be this?

Reply:
No, as it would copy to same `'option'` key for every `option` value, i.e. `x.option = y.cacheDir` instead of `x.cacheDir = y.cacheDir`